### ISSUE TITLE
Support URI Template expansion

### ISSUE THREAD

**Issue by gmlewis** [google/google-api-go-client#65]
**tmatsuo@google.com** on 23 Jul 2014 at 8:26:

```

Currently the client library doesn't handle forms like follows:
"topics/{+topic}"

For example, Cloud Pub/Sub API has such methods and the generated library does 
not work correctly.

FYI, the discovery doc is here:
https://www.googleapis.com/discovery/v1/apis/pubsub/v1beta1/rest
```

**Comment by gmlewis** [comment]
**tmatsuo@google.com** on 24 Jul 2014 at 12:58:

```
Here is the RFC for URI Template:
http://tools.ietf.org/html/rfc6570

Pub/Sub API has a method topics.delete defined as follows (see "path" which 
expects URI Template expansion):
"delete": {
     "id": "pubsub.topics.delete",
     "path": "topics/{+topic}",
     "httpMethod": "DELETE",
     "description": "Deletes the topic with the given name. All subscriptions to this topic are also deleted. Returns NOT_FOUND if the topic does not exist. After a topic is deleted, a new topic may be created with the same name.",
     "parameters": {
      "topic": {
       "type": "string",
       "description": "Name of the topic to delete.",
       "required": true,
       "location": "path"
      }
     },
     "parameterOrder": [
      "topic"
     ],
     "scopes": [
      "https://www.googleapis.com/auth/cloud-platform",
      "https://www.googleapis.com/auth/pubsub"
     ]
    },
```

**Comment by gmlewis** [comment]
**bradfitz@google.com** on 18 Aug 2014 at 7:14:

```
This issue was closed by revision 11e893087017.
```

_Status changed to **Fixed**._
